src/sounds.cpp

```diff
@@ -388,16 +393,790 @@ std::pair<std::vector<tripoint>, std::vector<tripoint>> sounds::get_monster_soun
+                    g->weather != previous_weather && !g->u.get_effect_int( "deaf" ) > 0 ) ) {
+        fade_audio_group( 1, 1000 );
+        fade_audio_group( 2, 1000 );
+        play_ambient_variant_sound( "environment", "underground", get_heard_volume( g->u.pos() ), 2,
+                                    1000 );
+        // We are indoors
+    } else if( ( g->is_sheltered( g->u.pos() ) && !g->is_underground( g->u.pos() ) &&
+                 is_channel_playing( 3 ) != 1 && !g->u.get_effect_int( "deaf" ) > 0 ) ||
+               ( g->is_sheltered( g->u.pos() ) && !g->is_underground( g->u.pos() ) &&
+                 g->weather != previous_weather && !g->u.get_effect_int( "deaf" ) > 0 ) ) {
+        fade_audio_group( 1, 1000 );
+        fade_audio_group( 2, 1000 );
+        play_ambient_variant_sound( "environment", "indoors", get_heard_volume( g->u.pos() ), 3, 1000 );
+    }
+    // We are indoors and it is also raining
+    if( g->weather >= WEATHER_DRIZZLE && g->weather <= WEATHER_ACID_RAIN && !g->is_underground( g->u.pos() )
+            && is_channel_playing( 4 ) != 1 ) {
+        play_ambient_variant_sound( "environment", "indoors_rain", get_heard_volume( g->u.pos() ), 4,
+                                    1000 );
+    }
+    if( ( !g->is_sheltered( g->u.pos() ) && g->weather != WEATHER_CLEAR
+            && is_channel_playing( 5 ) != 1  &&
+            is_channel_playing( 6 ) != 1  && is_channel_playing( 7 ) != 1  && is_channel_playing( 8 ) != 1
+            &&
+            is_channel_playing( 9 ) != 1  && !g->u.get_effect_int( "deaf" ) > 0 )
+            || ( !g->is_sheltered( g->u.pos() ) &&
+                 g->weather != previous_weather  && !g->u.get_effect_int( "deaf" ) > 0 ) ) {
+        fade_audio_group( 1, 1000 );
+        // We are outside and there is precipitation
+        switch( g->weather ) {
+	case WEATHER_ACID_DRIZZLE:        
+	case WEATHER_DRIZZLE:
+            play_ambient_variant_sound( "environment", "WEATHER_DRIZZLE", get_heard_volume( g->u.pos() ), 9,
+                                        1000 );
+            break;
+        case WEATHER_RAINY:
+            play_ambient_variant_sound( "environment", "WEATHER_RAINY", get_heard_volume( g->u.pos() ), 8,
+                                        1000 );
+            break;
+	case WEATHER_ACID_RAIN:
+	case WEATHER_THUNDER:
+        case WEATHER_LIGHTNING:
+            play_ambient_variant_sound( "environment", "WEATHER_THUNDER", get_heard_volume( g->u.pos() ), 7,
+                                        1000 );
+            break;
+        case WEATHER_FLURRIES:
+            play_ambient_variant_sound( "environment", "WEATHER_FLURRIES", get_heard_volume( g->u.pos() ), 6,
+                                        1000 );
+            break;
+	// I didn't know where to put these or even if the first two were needed. They probably shouldn't make any noise, but I was getting an error when compiling when I left them out.
```

Comment:
It's fine as it is. The weather types NULL and NUM should never appear anyway.

If somebody adds a new weather type, the compiler will warn again that the new type is not handled here (and in every other similar `switch` statement) so one can not forget to update them all.